### PR TITLE
FFmpeg: replace deprecated av_init_packet API

### PR DESCRIPTION
**Pull request by JoelLinn**
`sizeof(AVPacket)` being a part of the public FFmpeg ABI is deprecated.
https://ffmpeg.org/doxygen/trunk/structAVPacket.html

This patch fixes building against FFmpeg 4.4 with `-Werror`
Due to my use case I am limited in testing this thoroughly though :/

### CODE REVIEW

**Comment by JoelLinn** [comment]
Fixes #1262 
@erankor

File: vod/filters/audio_encoder.c [hunk 1]
```diff
@@ -200,11 +200,15 @@ audio_encoder_write_frame(
...
+			"audio_encoder_write_frame: av_packet_alloc failed");
+		return VOD_ALLOC_FAILED;
+	}
+	// packet data will be allocated by the encoder
 
-	avrc = avcodec_receive_packet(state->encoder, &output_packet);
+	avrc = avcodec_receive_packet(state->encoder, output_packet);
 
 	if (avrc == AVERROR(EAGAIN))
 	{
```
Comment (erankor) [inline vod/filters/audio_encoder.c:214]: missing free here?

File: vod/filters/audio_encoder.c [hunk 2]
```diff
@@ -261,16 +270,16 @@ audio_encoder_flush(
 			return VOD_UNEXPECTED;
```
Comment (erankor) [inline vod/filters/audio_encoder.c:270]: missing free?

**Review by erankor — CHANGES_REQUESTED** [review]
Looks good, please check the couple of leaks that I mentioned.

**Comment by JoelLinn** [comment]
Good catches!
I added the two frees

**Comment by rhummelmose** [comment]
Lgtm :)

**Comment by r4fek** [comment]
Looks like all issues were resolved. Can we merge this PR?

**Comment by rhummelmose** [comment]
Maybe @erankor is on summer vacay :)

**Comment by erankor** [comment]
Merged, thanks!

**Pull request merged**